core/src/main/scala/filodb.core/metadata/Schemas.scala

```diff
@@ -0,0 +1,237 @@
+package filodb.core.metadata
+
+import com.typesafe.config.Config
+import net.ceedubs.ficus.Ficus._
+import org.scalactic._
+
+import filodb.core.binaryrecord2.{RecordBuilder, RecordComparator, RecordSchema}
+import filodb.core.downsample.ChunkDownsampler
+import filodb.core.query.ColumnInfo
+import filodb.core.store.ChunkSetInfo
+import filodb.core.Types._
+import filodb.memory.{BinaryRegion, MemFactory}
+import filodb.memory.format.BinaryVector
+
+/**
+ * A DataSchema describes the data columns within a time series - the actual data that would vary from sample to
+ * sample and is encoded.  It has a unique hash code for each unique DataSchema.
+ * One Dataset in FiloDB can comprise multiple DataSchemas.
+ * One DataSchema should be used for each type of metric or data, such as gauge, histogram, etc.
+ * The "timestamp" or rowkey is the first column and must be either a LongColumn or TimestampColumn.
+ * DataSchemas are intended to be built from config through Schemas.
+ */
+final case class DataSchema private(name: String,
+                                    columns: Seq[Column],
+                                    downsamplers: Seq[ChunkDownsampler],
+                                    hash: Int,
+                                    valueColumn: ColumnId) {
+  val timestampColumn  = columns.head
+  val timestampColID   = 0
+
+  // Used to create a `VectorDataReader` of correct type for a given data column ID;  type PtrToDataReader
+  val readers          = columns.map(col => BinaryVector.defaultPtrToReader(col.columnType.clazz)).toArray
+
+  // The number of bytes of chunkset metadata including vector pointers in memory
+  val chunkSetInfoSize = ChunkSetInfo.chunkSetInfoSize(columns.length)
+  val blockMetaSize    = chunkSetInfoSize + 4
+
+  def valueColName: String = columns(valueColumn).name
+}
+
+/**
+ * A PartitionSchema is the schema describing the unique "key" of each time series, such as labels.
+ * The columns inside PartitionSchema are used for distribution and sharding, as well as filtering and searching
+ * for time series during querying.
+ * There should only be ONE PartitionSchema across the entire Database.
+ */
+final case class PartitionSchema(columns: Seq[Column],
+                                 predefinedKeys: Seq[String],
+                                 options: DatasetOptions) {
+  val binSchema = new RecordSchema(columns.map(c => ColumnInfo(c.name, c.columnType)), Some(0), predefinedKeys)
+
+  private val partKeyBuilder = new RecordBuilder(MemFactory.onHeapFactory, binSchema, 10240)
```

Comment:
Magic value?

Reply:
Ah ok, it was copy and pasted.  Thanks for catch